pkg/config/config.go

```diff
@@ -46,6 +105,149 @@ type Config struct {
 	Bundles            Bundles           `yaml:"bundles,omitempty"`
 	GrubOptions        map[string]string `yaml:"grub_options,omitempty"`
 	Env                []string          `yaml:"env,omitempty"`
+	// From elemental
+	Debug                     bool     `yaml:"debug,omitempty" mapstructure:"debug"`
+	Strict                    bool     `yaml:"strict,omitempty" mapstructure:"strict"`
+	CloudInitPaths            []string `yaml:"cloud-init-paths,omitempty" mapstructure:"cloud-init-paths"`
+	EjectCD                   bool     `yaml:"eject-cd,omitempty" mapstructure:"eject-cd"`
+	FullCloudConfig           string   // Stores the full cloud config used to generate the spec afterwards
+	Logger                    v1.Logger
+	Fs                        v1.FS
+	Mounter                   mount.Interface
+	Runner                    v1.Runner
+	Syscall                   v1.SyscallInterface
+	CloudInitRunner           v1.CloudInitRunner
+	ImageExtractor            v1.ImageExtractor
+	Client                    v1.HTTPClient
+	Platform                  *v1.Platform `yaml:"platform,omitempty" mapstructure:"platform"`
+	Cosign                    bool         `yaml:"cosign,omitempty" mapstructure:"cosign"`
+	Verify                    bool         `yaml:"verify,omitempty" mapstructure:"verify"`
+	CosignPubKey              string       `yaml:"cosign-key,omitempty" mapstructure:"cosign-key"`
+	Arch                      string       `yaml:"arch,omitempty" mapstructure:"arch"`
+	SquashFsCompressionConfig []string     `yaml:"squash-compression,omitempty" mapstructure:"squash-compression"`
+	SquashFsNoCompression     bool         `yaml:"squash-no-compression,omitempty" mapstructure:"squash-no-compression"`
+}
+
+// WriteInstallState writes the state.yaml file to the given state and recovery paths
+func (c Config) WriteInstallState(i *v1.InstallState, statePath, recoveryPath string) error {
+	data, err := yaml.Marshal(i)
+	if err != nil {
+		return err
+	}
+
+	data = append([]byte("# Autogenerated file by elemental client, do not edit\n\n"), data...)
+
+	err = c.Fs.WriteFile(statePath, data, constants.FilePerm)
+	if err != nil {
+		return err
+	}
+
+	err = c.Fs.WriteFile(recoveryPath, data, constants.FilePerm)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// LoadInstallState loads the state.yaml file and unmarshals it to an InstallState object
+func (c Config) LoadInstallState() (*v1.InstallState, error) {
+	installState := &v1.InstallState{}
+	data, err := c.Fs.ReadFile(filepath.Join(constants.RunningStateDir, constants.InstallStateFile))
```

Comment:
I see this was copied from before so no need to change in this PR but it feels like the full function could be a method of config

Reply:
? isnt it already a method of Config?

Reply:
but not a big deal this one